dal_admin_filters/__init__.py

```diff
@@ -2,17 +2,18 @@
 from dal import autocomplete
 from django import forms
 from django.contrib.admin.filters import SimpleListFilter
+from django.contrib.admin.utils import get_fields_from_path
 from django.core.exceptions import ImproperlyConfigured
 from django.forms.widgets import Media, MEDIA_TYPES
 
 
 class AutocompleteFilter(SimpleListFilter):
     template = "dal_admin_filters/autocomplete-filter.html"
     title = ''
-    field_name = ''
-    field_pk = 'id'
+    parameter_name = ''
     autocomplete_url = ''
     is_placeholder_title = False
+    widget = autocomplete.ModelSelect2
```

Comment:
Regardless of the overall proposed changes in this PR, configurable widget is the must-have feature. For example, one has to override `autocomplete.ModelSelect2.filter_choices_to_render` if `ForeignKey` field points to a field that is not `pk` (e.g., `country = ForeignKey(Country, to_field='code'`) because a lookup is hardcoded: https://github.com/yourlabs/django-autocomplete-light/blob/14af18991894e1b45a1d711ea326fcfc6fc47509/src/dal/widgets.py#L188 (`pk__in`).